src/ostree/ot-admin-builtin-upgrade.c

```diff
@@ -95,9 +95,6 @@ ot_admin_builtin_upgrade (int argc, char **argv, GCancellable *cancellable, GErr
                                                   "override-commit", NULL);
         }
 
-      /* Should we consider requiring --discard-hotfix here? */
-      origin_changed |= g_key_file_remove_key (origin, "origin", "unlocked", NULL);
```

Comment:
The extra config options will be silently ignored, and preserve the option to downgrade to an older version of OSTree.  I'm not sure how deleting them helps backwards compatibility. 

I'm also not sure why backwards compatibility is a concern at all, at this point in time, but that's a separate issue.